### PR TITLE
Add secrets documentation

### PR DESCRIPTION
**Pull request by gciltaru**
#### Purpose
Add Secret documentation

### CODE REVIEW

File: api/reference/1.0/secrets/create.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+#### `succeeded`
+
+This is the status when the exchange request for the secret completes successfully.
+
+#### `failed`
+
+This is the status when the exchange request for the secret is not successful.
+
+{% form secret.post %}
```
Comment (thebenrob) [inline api/reference/1.0/secrets/create.md:21]: Can/should we provide examples of creating secrets of different types?  I'm assuming this example is an OAuth one, but what about creating tokens or username/password secrets?

File: api/reference/1.0/secrets/create.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+At the same time, a secret exchange task is triggered to test that the credential exchange is functional.  This task is processed asynchronously and updates the secret's `status` attribute with one of two possible values:
+
+#### `succeeded`
+
+This is the status when the exchange request for the secret completes successfully.
+
+#### `failed`
+
+This is the status when the exchange request for the secret is not successful.
+
```
Comment (thebenrob) [inline api/reference/1.0/secrets/create.md:20]: I think we should provide information to the user for failed secrets.  Where will they get information on what went wrong?

File: api/reference/1.0/secrets/relationships/environment/list_related.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+title: Fetch related
```
Comment (thebenrob) [inline api/reference/1.0/secrets/relationships/environment/list_related.md:2]: We also need to create a page for the Environments relationship to Secrets to show how an Environment can retrieve its related secrets

**Review by thebenrob — CHANGES_REQUESTED** [review]

File: api/reference/1.0/secrets/create.md (hunk 2 above)
Comment (gciltaru) [reply api/reference/1.0/secrets/create.md:20]: I added the `api/guides/secrets.md`. Can you check it out?

File: api/reference/1.0/secrets/create.md (hunk 1 above)
Comment (gciltaru) [reply api/reference/1.0/secrets/create.md:21]: Done

File: api/reference/1.0/secrets/relationships/environment/list_related.md (hunk 3 above)
Comment (gciltaru) [reply api/reference/1.0/secrets/relationships/environment/list_related.md:2]: Done

**Review by thebenrob — DISMISSED** [review]

**Comment by thebenrob** [comment]
I haven't built an image to test everything, but the content is good.

**Comment by brenthosie** [comment]
@pnolan89 This seems stuck 😢

**Comment by gciltaru** [comment]
@pnolan89 can you please help us with this stuck PR?

**Review by pnolan89 — DISMISSED** [review]
Sorry for the delay! The content looks complete. I should have everything I need to put together the Experience League docs for this info--approved.

Once this is merged I'll start migrating the content over and update for style and structure, tracking under the following Jira ticket: https://jira.corp.adobe.com/browse/PLAT-101972

**Review by remear — APPROVED** [review]

**Pull request merged**